src/config.ts

```diff
@@ -404,7 +408,11 @@ export class KubeConfig {
             opts.headers = [];
         }
         if (authenticator) {
-            await authenticator.applyAuthentication(user, opts);
+            const refresh = await authenticator.applyAuthentication(user, opts);
+            if (refresh && this.configFile.length > 0) {
+                const contents = this.exportConfig();
+                await fs.writeFile(this.configFile, contents, () => {});
```

Comment:
I have one concern about this - if config file was changed on disk, e.g. by using `kubectl` in another context, then this will discard any changes made there and can lead to the similar problem with stale refresh token. Since we know what has been changed, you could patch config file on disk, instead of exporting current config.

Reply:
This race is that this is why there is a `tokenFile` directive in the kubeconfig file. (I'm realizing actually that this PR isn't correct because it doesn't properly handle `tokenFile`

Perhaps I will change this so that it only does the refresh when `tokenFile` is present.

(concerns like this are why this library doesn't have refresh currently :)

Reply:
But in my case for example, I don't have `tokenFile`, but I still need token to be refreshed. 
Ok, maybe for now I will try to craft some custom solution which will track when config is changed by the client and merge those changes back into the kubeconfig file and see how that works.

Reply:
The trouble is that the file always could have changed underneath you between when you check and when you write. There's no way to guarantee that it won't occur.

Reply:
That is true. 

My idea was that I could maintain two copies of kubeconfig: the original one, which I loaded when application started, and a copy which is being updated by the library. Then I could periodically load kubeconfig from the filesystem, somehow make a diff with my original copy to see what has changed, do the same for the changes made by library. Then order changes by date, resolve conflicts and write back. 

The file needs to be locked during all this time to prevent any modification in-between, at least I think this is how it's done in Go client.

But yeah, that is not a trivial task :)

Reply:
What if instead of writing to a file it would fire some event instead, which could be handled by some other component? I've been working on this kind of workaround for my own purposes:

```js
import * as fs from 'fs';
import { KubeConfig, dumpYaml, loadYaml, findHomeDir } from '@kubernetes/client-node';

export class KubeConfigSynchronizer {
    private config: KubeConfig;
    private path: string = `${findHomeDir()}/.kube/config`;

    constructor(config: KubeConfig) {
        this.patch(); // patch KubeConfig to install update listeners
        this.config = config;

        fs.watch(this.path, (event, filename) => {
            config.loadFromDefault();
            console.log('Kube config reloaded');
        });
    }

    update() {
        const data = dumpYaml(loadYaml(this.config.exportConfig()));
        fs.writeFile(this.path, data, {}, () => {
            console.log("Kube config updated");
        });        
    }

    private patch() {
        const MyKubeConfig = KubeConfig as any
        MyKubeConfig.authenticators.forEach((authenticator: any) => {
            const applyAuthentication = authenticator.applyAuthentication;
            const newApplyAuthentication = (user: any, opts: any) => {
                return applyAuthentication.call(authenticator, user, opts).then((result: any) => {
                    this.update();
                    return result;
                }).catch((e: any) => {
                    throw e;
                });
            };
            authenticator.applyAuthentication = newApplyAuthentication.bind(authenticator);
        });
    }
}
```

It's not ideal right now, because sometimes `loadFromDefault()` throws an exception which I haven't looked into yet, but it would make the whole thing a bit easier if I could subscribe for kubeconfig update events, instead of patching authenticators.